data/mods/ssb/abilities.ts

```diff
@@ -51,6 +51,53 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 		},
 	},
 
+	// Cake
+  h: {
+		desc: "On switch-in and at the ened of every turn, this Pokemon changes to a random typing.",
+		shortDesc: "On switch-in & every turn, random type.",
+		name: "h",
+		onSwitchIn(pokemon) {
+			let typeList = ["Normal", "Fighting", "Flying", "Poison", "Ground", "Rock",
+								"Bug", "Ghost", "Steel", "Fire", "Water", "Grass", "Electric",
+								"Psychic", "Ice", "Dragon", "Dark", "Fairy"];
+			this.prng.shuffle(typeList);
+			let firstType = typeList[0];
+			this.prng.shuffle(typeList);
+			let secondType = typeList[0];
+			let newTypes = [];
+			if (firstType === secondType) {
+				newTypes = [firstType]
```

Comment:
oops

Reply:
What's the issue here? It's so that it doesn't show as Electric/Electric.

Reply:
```suggestion
				newTypes = [firstType];
```